File: app/src/main/java/baidu/com/testlibproject/service/VibratorActivity.java
package baidu.com.testlibproject.service;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.os.Vibrator;
import android.view.MotionEvent;
import android.widget.Toast;

import baidu.com.testlibproject.R;

public class VibratorActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.vibrator_layout);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_UP:
                Vibrator vibrator = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
                Toast.makeText(this, "vibrate!", Toast.LENGTH_SHORT).show();
                vibrator.vibrate(500);
                break;
        }
        return super.onTouchEvent(event);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: app/src/main/java/baidu/com/testlibproject/service/VibratorActivity.kt
package baidu.com.testlibproject.service

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.os.Vibrator
import android.view.MotionEvent
import android.widget.Toast

import baidu.com.testlibproject.R

class VibratorActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.vibrator_layout)
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_UP -> {
                val vibrator = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
                Toast.makeText(this, "vibrate!", Toast.LENGTH_SHORT).show()
                vibrator.vibrate(500)
            }
        }
        return super.onTouchEvent(event)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }
}
